Formula/um.rb

```diff
@@ -15,6 +15,10 @@ class Um < Formula
 
   depends_on "ruby" if MacOS.version <= :sierra
 
+  on_linux do
+    depends_on "ruby"
+  end
```

Comment:
Why not
```
uses_from_macos "ruby", since: :high_sierra
```
?

Reply:
Because `uses_from_macos "ruby"` does not do anything on Mac.

Reply:
We did the same in `asciidoctor`

Reply:
On macOS,
```ruby
uses_from_macos "ruby", since: :high_sierra
```
will resolve to
```
depends_on "ruby" if MacOS.version < :high_sierra
```
It's only a no-op when `MacOS.version >= :high_sierra`.

https://rubydoc.brew.sh/SoftwareSpec.html#uses_from_macos-instance_method

Reply:
I did not know that. Cool.